### PR TITLE
fix: DeckPicker onSaveInstanceState checks if fab is visible if not makes it visible

### PR DESCRIPTION
**Pull request by CyberButton**
## Purpose / Description
_Add button disappear after perform a search and go back_

## Fixes
* Fixes _https://github.com/ankidroid/Anki-Android/issues/15352_

## Approach
_onSaveInstanceState check if the FAB is visible, if not make it visible_

## How Has This Been Tested?

was tested manually

## Checklist
_Please, go through these checks before submitting the PR._

- [ x ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ x ] You have commented your code, particularly in hard-to-understand areas
- [ x ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by criticalAY — COMMENTED** [review]
Tests failing 
```
DeckPickerImportTest > replaceShowsImportDialog FAILED
    kotlin.UninitializedPropertyAccessException: lateinit property floatingActionMenu has not been initialized
        at com.ichi2.anki.DeckPicker.onResume(DeckPicker.kt:1113)
        at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1531)
        at org.robolectric.android.internal.RoboMonitoringInstrumentation.callActivityOnResume(RoboMonitoringInstrumentation.java:395)
        at android.app.Activity.performResume(Activity.java:8422)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at org.robolectric.shadows._Activity_$$Reflector33.performResume(Unknown Source)
        at org.robolectric.android.controller.ActivityController.lambda$resume$5(ActivityController.java:197)
        at org.robolectric.shadows.ShadowPausedLooper.runPaused(ShadowPausedLooper.java:227)
        at org.robolectric.android.controller.ComponentController.invokeWhilePaused(ComponentController.java:63)
        at org.robolectric.android.controller.ActivityController.resume(ActivityController.java:192)
        at com.ichi2.anki.RobolectricTest$Companion.startActivityNormallyOpenCollectionWithIntent(RobolectricTest.kt:265)
        at com.ichi2.anki.RobolectricTest.startActivityNormallyOpenCollectionWithIntent(RobolectricTest.kt)
        at com.ichi2.anki.RobolectricTest.startActivityNormallyOpenCollectionWithIntent$AnkiDroid_playDebugUnitTest(RobolectricTest.kt:337)
        at com.ichi2.anki.DeckPickerImportTest.replaceShowsImportDialog(DeckPickerImportTest.kt:41)
```

**Comment by CyberButton** [comment]
> Tests failing
> 
> ```
> DeckPickerImportTest > replaceShowsImportDialog FAILED
>     kotlin.UninitializedPropertyAccessException: lateinit property floatingActionMenu has not been initialized
>         at com.ichi2.anki.DeckPicker.onResume(DeckPicker.kt:1113)
>         at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1531)
>         at org.robolectric.android.internal.RoboMonitoringInstrumentation.callActivityOnResume(RoboMonitoringInstrumentation.java:395)
>         at android.app.Activity.performResume(Activity.java:8422)
>         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
>         at org.robolectric.shadows._Activity_$$Reflector33.performResume(Unknown Source)
>         at org.robolectric.android.controller.ActivityController.lambda$resume$5(ActivityController.java:197)
>         at org.robolectric.shadows.ShadowPausedLooper.runPaused(ShadowPausedLooper.java:227)
>         at org.robolectric.android.controller.ComponentController.invokeWhilePaused(ComponentController.java:63)
>         at org.robolectric.android.controller.ActivityController.resume(ActivityController.java:192)
>         at com.ichi2.anki.RobolectricTest$Companion.startActivityNormallyOpenCollectionWithIntent(RobolectricTest.kt:265)
>         at com.ichi2.anki.RobolectricTest.startActivityNormallyOpenCollectionWithIntent(RobolectricTest.kt)
>         at com.ichi2.anki.RobolectricTest.startActivityNormallyOpenCollectionWithIntent$AnkiDroid_playDebugUnitTest(RobolectricTest.kt:337)
>         at com.ichi2.anki.DeckPickerImportTest.replaceShowsImportDialog(DeckPickerImportTest.kt:41)
> ```

oh I see, sry I'll fix that

**Comment by CyberButton** [comment]
Kindly request your assistance with reviewing @criticalAY

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 1]
```diff
@@ -1120,6 +1120,7 @@ open class DeckPicker :
         // As `loadDeckCounts` is cancelled in `migrate()`
         val message = dialogHandler.popMessage()
         super.onResume()
+
```
Comment (criticalAY) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1123]: Remove the extra line you added here

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 2]
```diff
@@ -1152,6 +1153,9 @@ open class DeckPicker :
         }
         exportingDelegate.onSaveInstanceState(savedInstanceState)
         savedInstanceState.putSerializable("mediaUsnOnConflict", mediaUsnOnConflict)
+
+        // Show the floating action button if it is hidden
```
Comment (criticalAY) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1157]: Get rid of this comment here

**Review by criticalAY — COMMENTED** [review]
Rest looks good

**Review by criticalAY — APPROVED** [review]
Thanks!

**Review by david-allison — APPROVED** [review]
Looks good to me!

https://github.com/ankidroid/Anki-Android/labels/squash-merge

**Comment by BrayanDSO** [comment]
I don't have time to assess what is the place to put that, but `onSaveInstanceState`'s goal is to just do what the name says and then other lifecycle methods can restore the view state if necessary, so I feel that it shouldn't be there

**Review by lukstbit — APPROVED** [review]
I agree with Brayan that onSaveInstanceState() isn't the correct place to have this call.
Looking through the code it should probably be in onDeckClicked() but I'm not sure if this will interact with the normal deck selection so I'm fine with onSaveInstanceState(as it does restore the fab in this issues' specific scenario, if the activity is recreated in the background the fab is shown normally).

**Pull request merged**